src/qamqpmessage.cpp

```diff
@@ -108,6 +108,11 @@ QVariant QAmqpMessage::header(const QString &header, const QVariant &defaultValu
     return d->headers.value(header, defaultValue);
 }
 
+const QHash<QString, QVariant>& QAmqpMessage::headers() const
```

Comment:
Yes, indeed.

I often use "const Foo&" syntax for getter, because of old habits !

Here, even "QHash<QString, QVariant>" is sufficient.